core/src/test/java/tech/pegasys/web3signer/core/multikey/metadata/parser/YamlSignerParserTest.java

```diff
@@ -271,16 +275,118 @@ void azureSecretMetadataInfoReturnsMetadata() throws IOException {
 
     final List<ArtifactSigner> result = signerParser.parse(filename);
     assertThat(result).containsOnly(artifactSigner);
-    verify(blsArtifactSignerFactory).create(hasCorrectAzureMetadataArguments());
+    verify(blsArtifactSignerFactory)
+        .create(hasCorrectAzureMetadataArguments(AzureAuthenticationMode.CLIENT_SECRET));
   }
 
-  private AzureSecretSigningMetadata hasCorrectAzureMetadataArguments() {
+  @ParameterizedTest
+  @EnumSource(AzureAuthenticationMode.class)
+  void azureSecretMetadataWithAuthenticationModeReturnsMetadata(
```

Comment:
not quite sure if this is testing our code, or Jackson.
Its got wide ranging coverage (validators, factories etc.) :/ not sure.

Reply:
It primarily testing YamlSignerParser's parse method ... with specific metadata ... and mocks to ensure parse method has called appropriate logic after metadata is parsed!